src/dart/lib/decoder.dart

```diff
@@ -0,0 +1,894 @@
+        || (((color & 0x0000FF00) >> 8) > (color & 0x000000FF));
+  }
+  static bool isGradient(int color) => (color & 0x000080FF) == 0x00008000;
+  
+  Gradient RGBAToGradient(int color, { ValueChanged<String>? onDebugDescribe }) {
+    assert(isGradient(color));
+    int NSTOPS = (color >> 24) & 0x3F;
+    final int CBASE = (color >> 16) & 0x3F;
+    final TileMode tileMode;
+    switch (color & 0x00C00000) {
+      case 0x00000000: // None
+        tileMode = TileMode.decal;
+        break;
+      case 0x00400000: // Pad
+        tileMode = TileMode.clamp;
+        break;
+      case 0x00800000: // Reflect
+        tileMode = TileMode.mirror;
+        break;
+      case 0x00C00000: // Repeat
+        tileMode = TileMode.repeated;
+        break;
+      default:
+        throw StateError('unreachable');
+    }
+    final int NBASE = (color >> 8) & 0x3F;
+    final bool isRadial = (color & 0x00004000) == 0x00004000;
+    final List<int> rawColors = List<int>.generate(NSTOPS, (int index) => CREG[(CBASE + index) % kCREGLength]).toList(); // "Color stops" in IconVG
+    final List<double> colorStops = List<double>.generate(NSTOPS, (int index) => NREG[(NBASE + index) % kNREGLength]); // "Offset stops" in IconVG
+
+    // IconVG uses premultiplied alpha, which means that it loses the information required to do gradients through a fully-transparent color.
+    // We resynthesize this information here, assuming that 0x00000000 means fully-transparent-whatever-is-next.
+    // As part of this we sometimes have to double-up any fully-transparent stops since the colors on either side may be different.
+    // RGBAToColor (used below to generate the list of Colors) does not remultiply colors that have been affected by this code.
+    for (int index = 0; index < NSTOPS; index += 1) {
+      if (rawColors[index] == 0x00000000) {
+        if (index > 0) {
+          rawColors[index] = rawColors[index - 1] & 0xFFFFFF00;
+        }
+        if (index < NSTOPS - 1) {
+          if ((rawColors[index + 1] & 0xFFFFFF00) != rawColors[index]) {
+            rawColors.insert(index + 1, rawColors[index + 1] & 0xFFFFFF00);
+            colorStops.insert(index + 1, colorStops[index]);
+            index += 1;
+            NSTOPS += 1;
+          } else {
+            rawColors[index] = rawColors[index + 1] & 0xFFFFFF00;
+          }
+        }
+      }
+    }
+
+    final List<Color> colors = rawColors.map<Color>((int value) => RGBAToColor(value)).toList();
+    
+    double a = NREG[(NBASE - 6) % kNREGLength];
+    double b = NREG[(NBASE - 5) % kNREGLength];
+    final double c = NREG[(NBASE - 4) % kNREGLength];
+    double d = NREG[(NBASE - 3) % kNREGLength];
+    double e = NREG[(NBASE - 2) % kNREGLength];
+    final double f = NREG[(NBASE - 1) % kNREGLength];
+    assert(() {
+      if (onDebugDescribe != null) {
+        onDebugDescribe('gradient (NSTOPS=$NSTOPS, CBASE=$CBASE, NBASE=$NBASE, ${isRadial ? "radial" : "linear"}, $tileMode)');
+      }
```

Comment:
nit: I think the style usually used for this is `onDebugDescribe?.call(....)`

Reply:
oops, meant to remove all this debug logic.

(FWIW I personally try to avoid `call` as much as possible.)